libs/sdk-core/src/greenlight.rs

```diff
@@ -306,6 +306,23 @@ impl NodeAPI for Greenlight {
         })
     }
 
+    async fn lookup_payment(&self, payment_hash: Vec<u8>) -> Result<Option<crate::Payment>> {
```

Comment:
Perhpas it is better to exten the sync process to also synchronize pending payments to the db and then just use the db existing lookpu for payment by hash?
The advantage is that we won't need to add another method to the NodeAPI trait and won't need to pass it as a dependency to the reverse swapper. Also we need the pending payments in the db as need to return them in the list_payments API.

Reply:
@roeierez  That's true. But if we do it that way, then we have to do a `sync` before checking a rev swap payment status, in case it's in progress, to get the latest payment state from GL.

For my test wallet, `sync` takes 2.8-3 seconds, `lookup_payment` about 0.7 seconds. Is that an acceptable delay?

Should I then keep using `lookup_payment` to determine the rev swap status, and inject / update this pending payment in our DB?

Or should I just add pending payments to `sync` and switch to using that, despite the extra delay?

Reply:
The extra delay is because the sync Api and the connection pooling are not optimal. This will change in the future so I think it is cleaner to use sync and query theocal db despite the extra temporary delay.
Let's make sure though that we call sync once for all addreses that we refresh and not once per address

Reply:
Done: d887e9deae88bf0810995c0653a7ee7442b84c78